src/main/java/demon/Commands.java

```diff
@@ -0,0 +1,285 @@
+     * If the task was marked as 'done', then executing this method will not change anything.
+     *
+     * @param tasks an array containing all tasks.
+     * @param num array index of the task list
+     */
+    public void mark(TaskList tasks, int num) {
+        if (num < 1 | num > tasks.getTaskList().size()) {
+            System.out.println(Ui.notWithinRange() + tasks.getTaskList().size());
+        } else {
+            Task item = tasks.getTaskList().get(num-1);
+            if (item.getStatusIcon().equals(" ")) {
+                item.markDone();
+                Ui.printDivider();
+                storage.reWriteFile(num);
+                System.out.println("Sure Master, I've marked this task as done :");
+                System.out.println(item);
+                Ui.printDivider();
+                Ui.promptNext();
+            } else {
+                System.out.println("Oops! Task already done!");
+            }
+        }
+    }
+
+    /**
+     * Creates a Deadline object as a task and stores into the taskList array.
+     * Method takes in the tasks arraylist and a string.
+     *
+     * @param tasks an array containing all tasks.
+     * @param input String input by the user containing the description, date and time.
+     * @throws NoTimingException if date and time not given.
+     * @throws EmptyDescriptionException if description not given.
+     * @throws IOException if fails to write new task into file.
+     * @throws DateTimeParseException if date and time given is invalid.
+     */
+    public void addDeadline(TaskList tasks, String input)
+            throws NoTimingException,
+                   EmptyDescriptionException,
+                   IOException,
+                   DateTimeParseException {
+        String[] parts = input.split(" ",2);
+        if (parts.length < 2) {
+            throw new EmptyDescriptionException("Description is EMPTY!");
+        }
+        String deadline = parts[1];
+        String[] parts2 = deadline.split("/by");
+        if (parts2.length < 2) {
+            throw new NoTimingException("WOI! Please include deadline!");
+        }
+        String description = parts2[0].trim();
+        String by = parts2[1].trim();
+
+        // Automatically assume that if time is not given, then time is 0000hrs
+        int sizeOfBy = by.split(" ").length;
+        if (sizeOfBy < 2) {
+            by += " 0000";
+        }
+        // Format the date, time, and create Deadline object, add to list
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+        LocalDateTime dateTime = LocalDateTime.parse(by, formatter);
+        Deadline itemDeadline = new Deadline(description, dateTime);
+        tasks.getTaskList().add(itemDeadline);
+        String stringToSave = "D | " + (itemDeadline.getStatusIcon().equals("X") ? "1" : "0")
+                            + " | " + description + " | " + by + "\n";
```

Comment:
Hi, can't I really see from here but I think the indentation for the wrapped line is more than 8 spaces 🤔 Small issue, can just take a quick look at your IDE to verify, hehe 😄